core/src/main/scala/api/sfs.scala

```diff
@@ -25,15 +30,19 @@ trait Filesystem {
    *  things too eagerly, nor caching data for too long.
    */
   def resolve(path: Path): Key
-  def metadata(key: Key): Metadata
-  def lookup(key: Key): Data
+  def metadata(key: Key): M[Metadata]
+  def lookup(key: Key): M[Data]
```

Comment:
If you don't have to touch filesystems when adding a new effect, you have invented magic. Effects aren't composable. There's no way to say a priori how they interact with existing effects.

But really I don't want to get lost in the weeds solving a problem at this level of generality. This thing can be written with zero type constructors and zero modeling of effects. That is good enough until such time as the lack of effect modeling becomes the obstacle. Because right now, the effect modeling is the obstacle.

In what sense are you saying fuse is blocking?